Usecase/usecase_transaction_test.go

```diff
@@ -1,157 +1,129 @@
+		mockTransactionRepo.On("SaveCreateTransaction", transaction).Return(transaction, nil).Once()
+		result, err := service.CreateTransaction(transaction)
 		assert.NoError(t, err)
+		assert.NotNil(t, result)
+		assert.Equal(t, transaction, result)
+		mockTransactionRepo.AssertExpectations(t)
 	})
 
 	t.Run("Invalid country", func(t *testing.T) {
-		TRepo := &mockTransactionRepo{
-			SaveCreateTransactionFunc: func(transaction *entities.Transaction) error {
-				return nil
-			},
-		}
-		PRepo := &mockProductRepo{
-			GetPriceProductsFunc: func(transaction *entities.Transaction) (*entities.Transaction, error) {
-				return &entities.Transaction{}, nil
-			},
-		}
-		service := NewTransactionUseCase(TRepo, PRepo)
-		err := service.CreateTransaction(
-			&entities.Transaction{
-				OrderAddress: "xx",
-				Items:        []entities.Item{{ProductId: 1, Quantity: 5}, {ProductId: 2, Quantity: 3}},
-			})
+		result, err := service.CreateTransaction(transactionInvalidCountry)
 		assert.Error(t, err)
+		assert.Nil(t, result)
 		assert.Equal(t, "dont have this country", err.Error())
+		mockTransactionRepo.AssertExpectations(t)
 	})
 
 	t.Run("Duplicate Id Product", func(t *testing.T) {
-		TRepo := &mockTransactionRepo{
-			SaveCreateTransactionFunc: func(transaction *entities.Transaction) error {
-				return nil
-			},
-		}
-		PRepo := &mockProductRepo{
-			GetPriceProductsFunc: func(transaction *entities.Transaction) (*entities.Transaction, error) {
-				return &entities.Transaction{}, nil
-			},
-		}
-		service := NewTransactionUseCase(TRepo, PRepo)
-		err := service.CreateTransaction(
-			&entities.Transaction{
-				OrderAddress: "th",
-				Items:        []entities.Item{{ProductId: 1, Quantity: 5}, {ProductId: 1, Quantity: 3}},
-			})
+		result, err := service.CreateTransaction(transactionDuplicateIdProduct)
 		assert.Error(t, err)
+		assert.Nil(t, result)
 		assert.Equal(t, "duplicate product_id", err.Error())
+		mockTransactionRepo.AssertExpectations(t)
 	})
 
 	t.Run("Cannot get PriceProducts", func(t *testing.T) {
-		TRepo := &mockTransactionRepo{
-			SaveCreateTransactionFunc: func(transaction *entities.Transaction) error {
-				return nil
-			},
-		}
-		PRepo := &mockProductRepo{
-			GetPriceProductsFunc: func(transaction *entities.Transaction) (*entities.Transaction, error) {
-				return nil, errors.New("cannot get price products")
-			},
-		}
-		service := NewTransactionUseCase(TRepo, PRepo)
-		err := service.CreateTransaction(&entities.Transaction{
-			OrderAddress: "th",
-			Items:        []entities.Item{{ProductId: 1, Quantity: 5}, {ProductId: 2, Quantity: 3}},
-		})
+		mockProductRepo.On("GetPriceProducts", transaction).Return(nil, errors.New("cannot get price products")).Once()
+		result, err := service.CreateTransaction(transaction)
 		assert.Error(t, err)
+		assert.Nil(t, result)
 		assert.Equal(t, "cannot get price products", err.Error())
+		mockTransactionRepo.AssertExpectations(t)
```

Comment:
You not have mockProductRepo.AssertExpectations(t) ?
why use  mockTransactionRepo.AssertExpectations(t) in case not use mockTransactionRepo.on()

Reply:
Fixed it. Please check again.